pkg/platform/kube/test/platform_test.go

```diff
@@ -49,6 +55,121 @@ type DeployFunctionTestSuite struct {
 	KubeTestSuite
 }
 
+func (suite *DeployFunctionTestSuite) TestDeployCronTriggerK8sWithJSONEventBody() {
+
+	// create an event-recorder function
+	functionPath := path.Join(suite.GetTestFunctionsDir(), "common", "event-recorder", "python", "event_recorder.py")
+	functionName := fmt.Sprintf("event-recorder-%s", xid.New().String())
+	createFunctionOptions := suite.CompileCreateFunctionOptions(functionName)
+
+	// get function source code
+	functionSourceCode, err := ioutil.ReadFile(functionPath)
+	suite.Require().NoError(err)
+
+	createFunctionOptions.FunctionConfig.Spec.Runtime = "python"
+	createFunctionOptions.FunctionConfig.Spec.Handler = "event_recorder:handler"
+	createFunctionOptions.FunctionConfig.Spec.Build.FunctionSourceCode = base64.StdEncoding.EncodeToString(functionSourceCode)
+
+	// set a JSON cron trigger event body
+	type myEventBody struct {
+		KeyA bool     `json:"key_a"`
+		KeyB []string `json:"key_b"`
+	}
+	cronTriggerEventBody := myEventBody{
+		KeyA: true,
+		KeyB: []string{
+			"value_1",
+			"value_2",
+		},
+	}
+	marshalledCronTriggerEventBody, err := json.Marshal(cronTriggerEventBody)
+	suite.Require().NoError(err)
+
+	createFunctionOptions.FunctionConfig.Spec.ServiceType = v1.ServiceTypeClusterIP
+	createFunctionOptions.FunctionConfig.Spec.Triggers = map[string]functionconfig.Trigger{
+		"crontrig": {
+			Kind: "cron",
+			Attributes: map[string]interface{}{
+				"interval": "5s",
+				"event": map[string]interface{}{
+					"body": string(marshalledCronTriggerEventBody),
+					"headers": map[string]interface{}{
+						"Extra-Header-1": "value1",
+						"Extra-Header-2": "value2",
+					},
+				},
+			},
+		},
+
+		// create http trigger with ingress, so we can invoke the function using it later (for testing on minikube)
+		"httptrig": {
+			Kind: "http",
+			Attributes: map[string]interface{}{
+				"ingresses": map[string]interface{}{
+					"0": map[string]interface{}{
+						"host":  testk8s.GetDefaultIngressHost(),
+						"paths": []string{"/"},
+					},
+				},
+			},
+		},
+	}
+
+	suite.DeployFunction(createFunctionOptions, func(deployResult *platform.CreateFunctionResult) bool {
+		var events []triggertest.Event
+
+		// wait for function to become ready
+		suite.WaitForFunctionState(&platform.GetFunctionsOptions{
+			Name:      functionName,
+			Namespace: suite.Namespace,
+		}, functionconfig.FunctionStateReady, time.Minute)
+
+		err = common.RetryUntilSuccessful(60*time.Second, 2*time.Second, func() bool {
+
+			// set http request url of the function
+			url := fmt.Sprintf("http://%s", testk8s.GetDefaultIngressHost())
```

Comment:
perhaps you can move this entire function as kube suite  helper of `suite.InvokeFunctionAndUnmarshal(url, interface\events)`

then from outside, you can retry until successful